src/zope/app/apidoc/utilities.py

```diff
@@ -105,12 +128,22 @@ def getPythonPath(obj):
     # accessed (which is probably not a bad idea). So, we remove the security
     # proxies for this check.
     naked = removeSecurityProxy(obj)
-    if hasattr(naked, "im_class"):
+    qname = ''
+    if hasattr(naked, '__qualname__'):
+        # Python 3. This makes unbound functions inside classes
+        # do the same thing as they do an Python 2: return just their
+        # class name.
+        qname = naked.__qualname__
+        qname = qname.split('.')[0]
+    if hasattr(naked, 'im_class'):
+        # Python 2, unbound methods
         naked = naked.im_class
+    if isinstance(naked, types.MethodType):
+        naked = type(naked.__self__)
```

Comment:
Why not `naked.__self__.__class__`?  I don't think `type()` will do the right thing for old-style classes on Python 2.
Then again `im_class` should've already taken care of this, for bound and unbound methods, on Python 2.  And Python 3 has `__qualname__`.  I'm not following why we need `__self__` at all...